dashboard/app/models/user.rb

```diff
@@ -2261,6 +2263,40 @@ def providers
     end
   end
 
+  # Returns number of times a user has attempted to join a section in the last 24 hours
+  # Returns 0 if no section join attempts
+  def num_section_attempts
+    section_attempts || 0
+  end
+
+  # There are two possible states in which we would want to reset section attempts
+  # 1) Initialize for the first time 2) 24 hours have passed since last attempt
```

Comment:
nit: I think this might be a bit clearer if this said `since last reset` instead of `since last attempt`